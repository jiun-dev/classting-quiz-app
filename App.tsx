import React from 'react';
import AppNavigator from './src/navigation/AppNavigator';
import {GestureHandlerRootView} from 'react-native-gesture-handler';

function App() {
  return (
    <GestureHandlerRootView style={{flex: 1}}>
      <AppNavigator />
    </GestureHandlerRootView>
  );
}

export default App;
