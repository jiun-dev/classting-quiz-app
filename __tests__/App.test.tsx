/**
 * @format
 */

import 'react-native';
import React, {ReactElement} from 'react';
import App from '../App';

// Note: import explicitly to use the types shiped with jest.
import {it} from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {render} from '@testing-library/react-native';

jest.mock('react-native-gesture-handler', () => ({
  ...jest.requireActual('react-native-gesture-handler'),
  GestureHandlerRootView: 'View',
}));

it('renders correctly', () => {
  renderer.create(<App />);
});

let props;
let component: React.ReactElement;

function getTempComponent(props: any): React.ReactElement {
  return <App {...props} />;
}

// test 그룹핑
describe('App 컴포넌트 테스트', () => {
  props = {};
  component = getTempComponent(props);
  // test == it(readable)
  test('컴포넌트 렌더링 테스트', () => {
    const rendered = render(component);
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });
});
