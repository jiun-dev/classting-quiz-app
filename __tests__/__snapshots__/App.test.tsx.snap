// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 컴포넌트 테스트 컴포넌트 렌더링 테스트 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "rgb(242, 242, 242)",
              "flex": 1,
            },
            undefined,
          ]
        }
      >
        <RNSScreenContainer
          onLayout={[Function]}
          style={
            {
              "flex": 1,
            }
          }
        >
          <RNSScreen
            activityState={2}
            collapsable={false}
            gestureResponseDistance={
              {
                "bottom": -1,
                "end": -1,
                "start": -1,
                "top": -1,
              }
            }
            onGestureCancel={[Function]}
            pointerEvents="box-none"
            sheetAllowedDetents="large"
            sheetCornerRadius={-1}
            sheetExpandsWhenScrolledToEdge={true}
            sheetGrabberVisible={false}
            sheetLargestUndimmedDetent="all"
            style={
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "opacity": 1,
                }
              }
            />
            <View
              accessibilityElementsHidden={false}
              closing={false}
              gestureVelocityImpact={0.3}
              importantForAccessibility="auto"
              onClose={[Function]}
              onGestureBegin={[Function]}
              onGestureCanceled={[Function]}
              onGestureEnd={[Function]}
              onOpen={[Function]}
              onTransition={[Function]}
              pointerEvents="box-none"
              style={
                [
                  {
                    "display": "flex",
                    "overflow": undefined,
                  },
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                ]
              }
              transitionSpec={
                {
                  "close": {
                    "animation": "spring",
                    "config": {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 10,
                      "restSpeedThreshold": 10,
                      "stiffness": 1000,
                    },
                  },
                  "open": {
                    "animation": "spring",
                    "config": {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 10,
                      "restSpeedThreshold": 10,
                      "stiffness": 1000,
                    },
                  },
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "flex": 1,
                    "marginTop": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  handlerTag={2}
                  handlerType="PanGestureHandler"
                  needsOffscreenAlphaCompositing={false}
                  onGestureHandlerEvent={[Function]}
                  onGestureHandlerStateChange={[Function]}
                  style={
                    {
                      "flex": 1,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                        {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    pointerEvents="box-none"
                    style={
                      [
                        {
                          "flex": 1,
                          "overflow": "hidden",
                        },
                        [
                          {
                            "backgroundColor": "rgb(242, 242, 242)",
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    <View
                      style={
                        {
                          "flex": 1,
                          "flexDirection": "column-reverse",
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "flex": 1,
                          }
                        }
                      >
                        <RCTSafeAreaView
                          style={
                            {
                              "alignItems": "center",
                              "flex": 1,
                              "justifyContent": "center",
                            }
                          }
                        >
                          <Text
                            style={
                              {
                                "color": "#4CAF50",
                                "fontSize": 24,
                                "fontWeight": "bold",
                                "marginBottom": 16,
                              }
                            }
                          >
                            CLASSTING QUIZ
                          </Text>
                          <View
                            style={
                              {
                                "marginVertical": 8,
                                "width": "80%",
                              }
                            }
                          >
                            <View
                              accessibilityState={
                                {
                                  "busy": undefined,
                                  "checked": undefined,
                                  "disabled": undefined,
                                  "expanded": undefined,
                                  "selected": undefined,
                                }
                              }
                              accessibilityValue={
                                {
                                  "max": undefined,
                                  "min": undefined,
                                  "now": undefined,
                                  "text": undefined,
                                }
                              }
                              accessible={true}
                              collapsable={false}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                {
                                  "alignItems": "center",
                                  "backgroundColor": "#4CAF50",
                                  "borderRadius": 16,
                                  "fontSize": 18,
                                  "marginVertical": 8,
                                  "opacity": 1,
                                  "paddingHorizontal": 24,
                                  "paddingVertical": 16,
                                }
                              }
                            >
                              <Text
                                style={
                                  {
                                    "color": "#FFFFFF",
                                    "fontSize": 22,
                                  }
                                }
                              >
                                퀴즈 풀기
                              </Text>
                            </View>
                          </View>
                          <View
                            style={
                              {
                                "marginVertical": 8,
                                "width": "80%",
                              }
                            }
                          >
                            <View
                              accessibilityState={
                                {
                                  "busy": undefined,
                                  "checked": undefined,
                                  "disabled": undefined,
                                  "expanded": undefined,
                                  "selected": undefined,
                                }
                              }
                              accessibilityValue={
                                {
                                  "max": undefined,
                                  "min": undefined,
                                  "now": undefined,
                                  "text": undefined,
                                }
                              }
                              accessible={true}
                              collapsable={false}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                {
                                  "alignItems": "center",
                                  "backgroundColor": "#4CAF50",
                                  "borderRadius": 16,
                                  "fontSize": 18,
                                  "marginVertical": 8,
                                  "opacity": 1,
                                  "paddingHorizontal": 24,
                                  "paddingVertical": 16,
                                }
                              }
                            >
                              <Text
                                style={
                                  {
                                    "color": "#FFFFFF",
                                    "fontSize": 22,
                                  }
                                }
                              >
                                오답 노트 확인
                              </Text>
                            </View>
                          </View>
                        </RCTSafeAreaView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RNSScreen>
        </RNSScreenContainer>
        <View
          collapsable={false}
          pointerEvents="box-none"
          style={
            {
              "height": 44,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 1,
            }
          }
        />
      </View>
    </RNCSafeAreaProvider>
  </View>
</View>
`;
